### PR TITLE
Replace Streamable w/ Writeable in BaseTasksResponse and subclasses

### PR DESCRIPTION
**Pull request by martijnvg**
This commit replaces usages of Streamable with Writeable for the
BaseTasksResponse / TransportTasksAction classes and subclasses of
these classes.

Note that where possible response fields were made final and default
constructors were removed. Also `TransportTasksAction#readTaskResponse(...)`
was no longer needed.

Relates to #34389

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/ListTasksResponse.java [hunk 1]
```diff
@@ -52,22 +52,28 @@
...
         this.tasks = tasks == null ? Collections.emptyList() : Collections.unmodifiableList(new ArrayList<>(tasks));
     }
 
+    public ListTasksResponse(StreamInput in) throws IOException {
+        super(in);
+        tasks = Collections.unmodifiableList(in.readList(TaskInfo::new));
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/ListTasksResponse.java]: :heart:

File: server/src/main/java/org/elasticsearch/action/support/tasks/TransportTasksAction.java [hunk 2]
```diff
@@ -71,20 +70,23 @@
 
     protected final ClusterService clusterService;
     protected final TransportService transportService;
-    protected final Writeable.Reader<TasksRequest> requestSupplier;
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/action/support/tasks/TransportTasksAction.java]: :+1:

File: server/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TransportTasksActionTests.java [hunk 3]
```diff
@@ -251,12 +245,7 @@ protected TestTasksResponse newResponse(TestTasksRequest request, List<TestTaskR
             return new TestTasksResponse(tasks, taskOperationFailures, failedNodeExceptions);
         }
 
-        @Override
-        protected TestTaskResponse readTaskResponse(StreamInput in) throws IOException {
-            return new TestTaskResponse(in);
-        }
-
-    }
+     }
```
Comment (nik9000) [inline server/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TransportTasksActionTests.java]: Extra space?

**Review by nik9000 — APPROVED** [review]
LGTM

**Comment by martijnvg** [comment]
run the gradle build tests 2

**Pull request merged**